tests/Schema/SchemaTest.php

```diff
@@ -33,6 +35,17 @@ public function testAddTable(): void
         self::assertTrue($schema->hasTable($tableName));
     }
 
+    public function testGetTableNames(): void
+    {
+        $conn = DriverManager::getConnection(TestUtil::getConnectionParams());
+
+        $schema = new Schema([], [], $conn->createSchemaManager()->createSchemaConfig());
+        $schema->createTable('foo');
+
+        $tableNames = $schema->getTableNames();
+        self::assertEquals(['foo'], $tableNames);
```

Comment:
The same call to `getTableNames()` on `2.12.1` and `2.13.x` returns `public.foo` (which is pointless given that it's SQLite). Where does the expectation of the unqualified `foo` come from?

It looks like this API has been broken since the beginning.

Reply:
I believe the proper approach would be to iterate over the schema tables and get their names. The array keys are meant to be used for indexing the tables within the schema, not used as names.

I'm thinking of fixing and deprecating this method.

Reply:
Fixed in https://github.com/doctrine/dbal/pull/4811.